### PR TITLE
Add evaluateTransactionExecutionUnitsWithLogs

### PR DESCRIPTION
**Pull request by ghost**
As we have switched to cardano-ledger for validation in plutus-apps, we also want the logs to get the coverage data. This PR adds `evaluateTransactionExecutionUnitsWithLogs` — a more general version of `evaluateTransactionExecutionUnits` with logs.

### CODE REVIEW

File: eras/alonzo/impl/src/Cardano/Ledger/Alonzo/Tools.hs [hunk 1]
```diff
@@ -185,7 +222,7 @@ evaluateTransactionExecutionUnits pp tx utxo ei sysS costModels = do
         (logs, Left e) -> case lang of
           PlutusV1 -> Left $ ValidationFailedV1 e logs
           PlutusV2 -> Left $ ValidationFailedV2 e logs
-        (_, Right exBudget) -> note (IncompatibleBudget exBudget) $ exBudgetToExUnits exBudget
+        (logs, Right exBudget) -> note (IncompatibleBudget exBudget) $ (logs,) <$> exBudgetToExUnits exBudget
```
Comment (lehins) [inline eras/alonzo/impl/src/Cardano/Ledger/Alonzo/Tools.hs]: No need to enable `TupleSections`
```suggestion
        (logs, Right exBudget) -> note (IncompatibleBudget exBudget) $ (,) logs <$> exBudgetToExUnits exBudget
```

File: eras/alonzo/impl/src/Cardano/Ledger/Alonzo/Tools.hs [hunk 2]
```diff
@@ -1,11 +1,13 @@
 {-# LANGUAGE DataKinds #-}
 {-# LANGUAGE FlexibleContexts #-}
 {-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TupleSections #-}
```
Comment (lehins) [inline eras/alonzo/impl/src/Cardano/Ledger/Alonzo/Tools.hs]: Redundant extension
```suggestion
```

**Review by lehins — APPROVED** [review]
LGTM. Minor suggestion to avoid `TupleSections`

**Comment by lehins** [comment]
Oh, yeah. You also need to run `scripts/ormolise.sh` in order to make the CI happy

**Review by JaredCorduan — APPROVED** [review]
this looks great, thank you @ak3n ! Let's add this to the changelog (the `Add` section of the `Unreleased`).

**Comment by ghost** [comment]
Thanks for your review! I want to finish the integration on our side at first, to verify that this implementation works for us. Will update the PR soon.

**Comment by ghost** [comment]
@lehins @JaredCorduan well, the CI is green.

Is it possible to backport it to `release/1.0.x` branch?

**Comment by JaredCorduan** [comment]
> Is it possible to backport it to `release/1.0.x` branch?

What is your goal, to get this in an earlier release?

**Comment by ghost** [comment]
>What is your goal, to get this in an earlier release?

At the moment we use [`c7c63dabdb215ebdaed8b63274965966f2bf408f`](https://github.com/input-output-hk/cardano-ledger/commit/c7c63dabdb215ebdaed8b63274965966f2bf408f) in our `cabal.project` and I would like to start using this function asap. Usually we track the dependencies of `cardano-wallet`.

I suppose the `master` went too far already.

**Comment by JaredCorduan** [comment]
> At the moment we use [`c7c63dabdb215ebdaed8b63274965966f2bf408f`](https://github.com/input-output-hk/cardano-ledger/commit/c7c63dabdb215ebdaed8b63274965966f2bf408f) in our `cabal.project` and I would like to start using this function asap. Usually we track the dependencies of `cardano-wallet`.

I feel like it would be confusing to extend `release/1.0.x` with a commit that we don't think will ever be a part of a release. The release process is all being moving to CHaPs at the moment, so things are a bit chaotic, sorry. In the short term, is there much difference for you between back-porting to `release/1.0.x` and just using a different branch? I'm open to suggestions? @lehins - do you have an opinion here?

**Comment by ghost** [comment]
>In the short term, is there much difference for you between back-porting to release/1.0.x and just using a different branch?

That works as well! I guess it's up to you how to orchestrate it — it's all about to use `cardano-ledger` in our `cabal.project` instead of a fork.

You can create `release/1.0.x-plutus-tools` based on `release/1.0.x` for example (or something else) and we can depend on that for now.

**Comment by JaredCorduan** [comment]
> You can create `release/1.0.x-plutus-tools` based on `release/1.0.x` for example (or something else) and we can depend on that for now.

Sounds good! I've made the branch.

We should squash your commits into one commit, and rebase. (I think GH can do this for us with one button). Then you can put up a PR against `release/1.0.x-plutus-tools`.

**Pull request merged**